packages/core/src/storage/IndyStorageService.ts

```diff
@@ -88,115 +91,147 @@ export class IndyStorageService<T extends BaseRecord> implements StorageService<
 
     try {
-      await this.wallet.updateWalletRecordValue(record.type, record.id, value)
-      await this.wallet.updateWalletRecordTags(record.type, record.id, tags)
+      await this.indy.updateWalletRecordValue(this.wallet.walletHandle, record.type, record.id, value)
+      await this.indy.updateWalletRecordTags(this.wallet.walletHandle, record.type, record.id, tags)
     } catch (error) {
       // Record does not exist
       if (isIndyError(error, 'WalletItemNotFound')) {
         throw new RecordNotFoundError(`record with id ${record.id} not found.`, {
           recordType: record.type,
           cause: error,
         })
-      } // Other indy error
-      else if (isIndyError(error)) {
-        handleIndyError(error)
       }
 
-      throw error
+      throw isIndyError(error) ? new IndySdkError(error) : error
     }
   }
 
-  /** @inheritDoc {StorageService#delete} */
+  /** @inheritDoc */
   public async delete(record: T) {
     try {
-      await this.wallet.deleteWalletRecord(record.type, record.id)
+      await this.indy.deleteWalletRecord(this.wallet.walletHandle, record.type, record.id)
     } catch (error) {
       // Record does not exist
       if (isIndyError(error, 'WalletItemNotFound')) {
         throw new RecordNotFoundError(`record with id ${record.id} not found.`, {
           recordType: record.type,
           cause: error,
         })
-      } // Other indy error
-      else if (isIndyError(error)) {
-        handleIndyError(error)
       }
 
-      throw error
+      throw isIndyError(error) ? new IndySdkError(error) : error
     }
   }
 
-  /** @inheritDoc {StorageService#getById} */
+  /** @inheritDoc */
   public async getById(recordClass: BaseRecordConstructor<T>, id: string): Promise<T> {
     try {
-      const record = await this.wallet.getWalletRecord(recordClass.type, id, IndyStorageService.DEFAULT_QUERY_OPTIONS)
+      const record = await this.indy.getWalletRecord(
+        this.wallet.walletHandle,
+        recordClass.type,
+        id,
+        IndyStorageService.DEFAULT_QUERY_OPTIONS
+      )
       return this.recordToInstance(record, recordClass)
     } catch (error) {
       if (isIndyError(error, 'WalletItemNotFound')) {
         throw new RecordNotFoundError(`record with id ${id} not found.`, {
           recordType: recordClass.type,
           cause: error,
         })
-      } else if (isIndyError(error)) {
-        handleIndyError(error)
       }
 
-      throw error
+      throw isIndyError(error) ? new IndySdkError(error) : error
     }
   }
 
-  /** @inheritDoc {StorageService#getAll} */
+  /** @inheritDoc */
   public async getAll(recordClass: BaseRecordConstructor<T>): Promise<T[]> {
-    const recordIterator = await this.wallet.search(recordClass.type, {}, IndyStorageService.DEFAULT_QUERY_OPTIONS)
+    const recordIterator = this.search(recordClass.type, {}, IndyStorageService.DEFAULT_QUERY_OPTIONS)
```

Comment:
Is the search not async any more? Are we missing an await?
```suggestion
    const recordIterator = await this.search(recordClass.type, {}, IndyStorageService.DEFAULT_QUERY_OPTIONS)
```

Reply:
`this.search` returns a generator, which we can use as an async iterator. So the `this.search` call itself is not async